mathics/builtin/files_io/filesystem.py

```diff
@@ -1350,7 +1350,10 @@ def eval_3(self, forms, paths, n, evaluation, **options):
         if options.get("System`IgnoreCase", None) is SymbolTrue:
             patterns = [
                 re.compile(
-                    "^" + to_regex(p, evaluation, abbreviated_patterns=True),
+                    "^"
+                    + to_regex(
```

Comment:
Note that there was a bug in both the original and here in that `to_regex()` can return `None` and then the string concatenation will raise an exception.

Reply:
OK, but what would be the best? return `None`, an empty string or to raise an exception?

Reply:
> OK, but what would be the best? return `None`, an empty string or to raise an exception?

This kind of thing has been debated a bit in programming languages. In Go and Rust they have gone the direction of returning `None` and the caller has to check. This is the more type-checked solution, because the compiler statically can detect whether this condition has been handled. 

In Python, C++ and many other languages the answer is looser. Python sometimes offers two methods, one that returns a special value and one that raises an error.  For example the `index` vesuss `find` methods on a string.